travis_ci.sh

```diff
@@ -30,25 +43,28 @@ function travis_run() {
 # Change to the directory this script is in
 cd $CURR_DIR
 
-# Note that we must build the codebase in order to run tests
-if [ "$RUN_BUILD" == "true" ] || [ "$RUN_TESTS" == "true" ]; then
+# Note that we must build the codebase in order to run tests and/or get coverage
+if [ "$RUN_BUILD" == "true" ] || \
+    [ "$RUN_TESTS" == "true" ] || \
+    [ "$RUN_COVERAGE" == "true" ]; then
     # Install all required dependecies
     travis_run ./environment_setup/setup_software.sh $ROS_DISTRO 
 
     # Build the codebase
-    travis_run catkin_make
+    travis_run catkin_make ${CMAKE_FLAGS}
 fi
 
 if [ "$RUN_TESTS" == "true" ]; then
-    # Run all the tests
-    travis_run catkin_make run_tests
+    travis_run catkin_make run_tests ${CMAKE_FLAGS}
 
     # Report the results of the tests
     # (which tests failed and why)
     travis_run catkin_test_results --verbose
 fi
 
-if [ "$RUN_FORMATTING_CHECKS" == "true" ]; then
+# We need to run tests in order to get coverage
+if [ "$RUN_FORMATTING_CHECKS" == "true" ] || \
+    [ "$RUN_COVERAGE" == "true" ]; then    
     CLANG_VERSION="7.0"
```

Comment:
(Unrelated to pull request)
I'm not too familiar with this kind of code, but maybe this should be defined somewhere else? It doesn't look like the clang version changes anywhere so maybe treat it like come kind of constant.

Reply:
Hmmm, yeah, I had it at the top of the file when I original wrote this, we might want to move it back there. @MathewMacDougall - did you have this here for a particular reason?

Reply:
Moving this conversation to slack so we can get this PR in.